### PR TITLE
Block dd-trace@3.42.0

### PR DESCRIPTION
**Pull request by 72636c**
This has the same memory leak as 4.21.0 (#102).

### CODE REVIEW

**Review by AaronMoat — APPROVED** [review]

File: third-party-major.json [hunk 1]
```diff
@@ -56,6 +56,12 @@
       "matchManagers": ["npm"],
       "matchPackageNames": ["dd-trace"],
 
+      "allowedVersions": "!/^3.42.0$/"
+    },
+    {
```
Comment (AaronMoat) [inline third-party-major.json:61]: Does having two separate rules and them together in the way we want here?

Comment (72636c) [reply third-party-major.json]: Ah great point - I've merged them together to be safe

**Review by AaronMoat — APPROVED** [review]

File: default.json [hunk 2]
```diff
@@ -79,7 +79,7 @@
       "matchManagers": ["npm"],
       "matchPackageNames": ["dd-trace"],
 
-      "allowedVersions": "!/^4.21.0$/"
+      "allowedVersions": "!/^(3.42.0|4.21.0)$/"
```
Comment (etaoins) [inline default.json:82]: Nitpick, but should these be e.g.:
```suggestion
      "allowedVersions": "!/^(3\\.42\\.0|4\\.21\\.0)$/"
```

Comment (72636c) [reply default.json:82]: I did look at this and couldn't think of how you'd defeat the regex between it enforcing boundaries `^$` and npm using semver.

But this is more correct 👍

**Pull request merged**